### ISSUE TITLE
Question: How to convert ExpPointerPoint to PointerPoint

### ISSUE THREAD

**Issue by alexserov** [microsoft/microsoft-ui-xaml#3674]
Hello,

I see that the WinUI3 Preview 3 introduced the following changes:
- the return type of the `Microsoft.UI.Xaml.Input.PointerRoutedEventArgs.GetCurrentPoint` method is now `Microsoft.UI.Input.Experimental.ExpPointerPoint`
- the return type of the `Microsoft.UI.Xaml.Input.PointerRoutedEventArgs.GetIntermediatePoints` method is now `Microsoft.UI.Input.Experimental.ExpPointerPoint`

However, I see no documentation related to the `ExpPointerPoint` class and no conversion operators or helper methods to get the `Windows.UI.Input.PointerPoint` instance from the ExpPointerPoint object.

Moreover, some of your API accepts the `PointerPoint` as parameters, e.g. the `Windows.UI.Input.GestureRecognizer.ProcessDownEvent` method.

---
So, my questions are:
- what is the proper way to convert `ExpPointerPoint` to `PointerPoint` and vice versa;
- what is the future of the `ExpPointerPoint`-related API? Do you plan to revert it? (I ask this question since I see that these types have similar vftbl)

**Comment by ranjeshj** [comment]
@jeffstall as FYI

**Comment by jeffstall** [comment]
Because WinUI / Project Reunion input operates differently than System Input, you cannot convert Microsoft.UI.Input.Experimental.ExpPointerPoint to a Windows.UI.Input.PointerPoint. Instead, it’s recommended to access the individual members of the ExpPointerPoint class and directly operate on the data.

With thanks!
~ Jeffrey

**Comment by alexserov** [comment]
Hey @jeffstall,

thanks for the answer. I understand that WinUI internals may differ from the rest framework, but it is difficult for me to understand why the public API reflects these differences in some cases.

As for my question, we use the `GestureRecognizer` class in the manner [described in the documentation](https://docs.microsoft.com/en-us/uwp/api/windows.ui.input.gesturerecognizer?view=winrt-19041) (see the Examples section). For now, this approach is broken, and, according to your answer, I conclude that we should not use it anymore.

After some experiments, I found out that we can make `GestureRecognizer` work by using the helper below, but I hope you have a proper solution of how to resolve such API conflicts so we could get rid of this workaround.

```csharp
  public static class PointerRoutedEventArgsExtensions {
        public static PointerPoint GetCurrentPointerPoint(this PointerRoutedEventArgs e, UIElement relativeTo) {
            return PointerPoint.GetCurrentPoint(e.Pointer.PointerId, new UIElementTransform(relativeTo));
        }
        public static IList<PointerPoint> GetIntermediatePointerPoints(this PointerRoutedEventArgs e, UIElement relativeTo) {
            return PointerPoint.GetIntermediatePoints(e.Pointer.PointerId, new UIElementTransform(relativeTo));
        }
        class GeneralPointerPointTransform : IPointerPointTransform {
            GeneralTransform generalTransform;
            public GeneralPointerPointTransform(GeneralTransform generalTransform) {
                this.generalTransform = generalTransform;
            }
            public bool TryTransform(Point inPoint, out Point outPoint) {
                outPoint = generalTransform.TransformPoint(inPoint);
                return true;
            }
            public Rect TransformBounds(Rect rect) {
                return generalTransform.TransformBounds(rect);
            }
            public IPointerPointTransform Inverse => new GeneralPointerPointTransform(this.generalTransform.Inverse);
        }
        class UIElementTransform : GeneralPointerPointTransform {
            GeneralTransform generalTransform;
            public UIElementTransform(UIElement source) : base(source?.TransformToVisual(null)?.Inverse ?? new MatrixTransform() { Matrix = Microsoft.UI.Xaml.Media.Matrix.Identity }) { }
        }
    }
```

**Comment by jlnordin** [comment]
@alexserov: Your workaround will work for the current preview release, but as @jeffstall said, the Reunion/WinUI input system works differently than system input, so this will likely break in future releases.

For now, you can also try the `Microsoft.UI.Input.Experimental.ExpGestureRecognizer` API that does take the `ExpPointerPoint` object. It is also not documented yet, but should follow pretty closely to the `Windows.UI.Input.GestureRecognizer` class. If there are issues with that, we'd like to know as we continue to improve this part of the WinUI / Reunion API surface.

**Comment by alexserov** [comment]
@jlnordin thanks for the info: I have not noticed that the `ExpGestureRecognizer` class exists. I'll let you know about the results once I check your advice on our side.

**Comment by michael-hawker** [comment]
We use the [`PointerPoint`](https://docs.microsoft.com/uwp/api/windows.ui.input.pointerpoint) a lot in the Windows Community Toolkit (including `DataGrid`). It seems that even though [`ExpPointerPoint`](https://docs.microsoft.com/windows/winui/api/microsoft.ui.input.experimental.exppointerpoint) is marked experimental, it's been shipped in the stable WASDK releases, like 0.8.x.

Will it still be shipped as part of 1.0 stable (or some equivalent API)? We want to know if we're going to be broken by this if there's a change or what alternative would be in place.

@ranjeshj @jeffstall any update/info on plans here for the near future? Thanks!
